src/Controls/Controls.js

```diff
@@ -35,11 +32,36 @@ class Controls extends Component {
     });
   }
 
-  render(props) {
-    let errorMessage = '';
-    if (this.props.isInvalidCardName) {
+  addCardToList = (cardName) => {
+    let cardNames = this.props.userCardsData.map(card => card.cardName);
+    if (!cardNames.includes(cardName)) {
+      this.setState({
+        hasDuplicates: false,
+        isInvalidCardName: false
+      });
+      this.props.addUserCard(cardName);
+    } else {
+      this.setState({ 
+        hasDuplicates: true,
+        isInvalidCardName: false
+      });
+    }
+  }
+
+  throwInvalidCardNameError = () => {
+    this.setState({ isInvalidCardName: true });
+  }
+
+  clearInput = () => {
+    this.textInput.current.value = '';
+    this.textInput.current.focus();
+  }
+
+  render() {
+    let errorMessage;
+    if (this.state.isInvalidCardName) {
       errorMessage = "The card name you entered is not in the database"
-    } else if (this.props.hasDuplicates) {
+    } else if (this.state.hasDuplicates) {
```

Comment:
The booleans used for error handling are now inside Control's state instead of being passed down from Aside.